### DISCUSSION TITLE
Speed Testing

### DISCUSSION THREAD

**Discussion by madmaximux** [chad-ermacora/rp-network-testers#13] [Ideas]
Hey! Thanks for replying at RPi forums. 

So my small suggestion is for option 3 that currently does nothing. I was thinking that it could be used to test the speed from the ISP. Separate from iperf testing, you'd know what your ISP was pushing you without having to get on your phone or PC and doing a speed test. So for me, when working in a tight networking closet, it would be super helpful to be able to do this without getting out and doing tests elsewhere  😛 . 

What I was thinking is that, option 3 would temporarily turn on dhcp, pull an IP from the router, and run a speed test against the ethernet, output the results, then disable dhcp again.  

One tool is [speedtest-cli](https://github.com/sivel/speedtest-cli), which can be used to specify the IP address of the ethernet adapter to bind to: `speedtest-cli --source 10.1.1.3` for example.

We can parse out info with either --simple or --json options:

```
➜   speedtest --simple
Ping: 15.839 ms
Download: 651.25 Mbit/s
Upload: 37.71 Mbit/s
```

What do you think? 

Edit: Grammar

**Comment by chad-ermacora** [comment]
That's a great idea. This is officially on my to-do list now. 

I'll have to read up on the Apache 2.0 license to make sure it will play nice with my GPL-3 but since it's already written in Python, it should be fairly easy to integrate. 

I'm currently finishing up a web interface overhaul of my other project 'Kootnet Sensors', but once I'm done that, I'll see about this.

**Reply by madmaximux** [reply]
Awesome! Let me know if I can help in any way. I haven’t looked at your code in detail yet, but can probably figure out where to drop this into with some trial and error.  

Also would their license be relevant if you are just downloading and running their tool only during installation time? Because that shouldn’t be an issue then. 

Or are you considering bundling it with the dep file? If you do that, you could drop their license file within the folder you keep it. 

Just thinking out loud. But to your point, it does seem to be compatible with GPL3.

**Reply by chad-ermacora** [reply]
Since it uses Python I'm pretty sure I can grab the main .py file and integrate that into the program, then I can access the functions directly in Python, removing the need to access the command line or do any additional installs. 

I'll know for sure when I take a closer look.

**Reply by madmaximux** [reply]
Neat!

**Comment by chad-ermacora** [comment]
I have uploaded a development build with the option to use SpeedTest.net (speedtest-cli). 

It's all built-in, so you should be able to just upgrade and run it (Make sure to do the Dev upgrade). I don't have it set up to show previous results yet, but you can run it and see the results of that test on the web interface and with the hardware E-Ink screen. 

Let me know how it goes. I'll add in other stuff like options and things later. Right now it's all automatic.

**Reply by madmaximux** [reply]
I just tried it out (sorry for the delay) and it works great and looks fantastic! Thank you for adding that!

Will there eventually be an option to test over ethernet only?

**Reply by chad-ermacora** [reply]
Yes, I will add additional options including the ability to change what networking device it runs over. I'm not sure when I'll get to it though ... I'll be sure to post the update here when It's been added.
